solver/cachemanager.go

```diff
@@ -217,6 +217,11 @@ func (c *cacheManager) LoadWithParents(ctx context.Context, rec *CacheRecord) ([
 			r.Release(context.TODO())
 		}
 	}
+	for _, r := range m {
```

Comment:
Shouldn't this be in the `defer` of `filterResults`?

Reply:
filterResults is recursive so it seemed complicated to try to update it relative to just emptying it out here. Figured it made sense still since m isn't constructed by filterResults